processing-service-core/api/swagger/swagger.yaml

```diff
@@ -952,7 +954,247 @@ paths:
+      tags:
+        - Tenants Configuration
+      summary: Retrieve effective tenant specific config.
+      description: Returns the effective tenant specific config for this key. That can be either a custom value specified for the tenant or the default.
+      operationId: getEffectiveTenantConfig
+      responses:
+        200:
+          description: Returns the effective tenant configurations for this key.
+          schema:
+            $ref: "#/definitions/effectiveTenantConfigValue"
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+  #
+  #
+  /tenants/{tenantId}/effectiveconfig:
+    x-swagger-router-controller: tenant
+    parameters:
+      - name: tenantId
+        in: path
+        type: string
+        description: The ID tenant.
+        required: true
+    get:
+      tags:
+        - Tenants Configuration
+      summary: Retrieve effective tenant specific config.
+      description: Returns the effective tenant specific config for this key. That can be either a custom value specified for the tenant or the default.
+      operationId: getEffectiveTenantConfigs
+      responses:
+        200:
+          description: Returns all effective configurations for this tenant.
+          schema:
+            $ref: "#/definitions/effectiveTenantConfigs"
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+  #
+  #
 definitions:
+  tenantConfigs:
+    type: object
+    properties:
+      configurations:
+        type: array
+        items:
+          $ref: "#/definitions/tenantConfig"
+        description: The collection of custom configurations set for the supplied tenantId.
+  tenantConfig:
+    type: object
+    properties:
+      key:
+        type: string
+        description: The key that used in retrieval of the custom config.
+      value:
+        type: string
+        description: The value stored for the custom config key.
+  effectiveTenantConfigs:
+    type: object
+    properties:
+      configurations:
+        type: array
+        items:
+          $ref: "#/definitions/effectiveTenantConfig"
+        description: The collection of custom configurations set for the supplied tenantId.
+  effectiveTenantConfig:
+    type: object
+    properties:
+      key:
+        type: string
+        description: The key that used in retrieval of the custom config.
+      value:
+        type: string
+        description: The value stored for the custom config key.
+      valueType:
+        type: string
+        description: The type of value being returned, CUSTOM if it was set specifically for this tenant, or DEFAULT if it cam from the global configs.
+        enum: ["CUSTOM", "DEFAULT"]
+  effectiveTenantConfigValue:
```

Comment:
You can avoid the repetition here by having effectiveTenantConfig use effetiveTenantConfigValue as part of its model.
e.g. in the existingCondition model in this contract it combines a property 'id' with another defined model using the below syntax;
```
allOf:
      - $ref: "#/definitions/condition"  
```